src/App.tsx

```diff
@@ -1,47 +1,111 @@
-import React from 'react';
-import './App.css';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+import React, { useState } from 'react';
+import './App.scss';
+import { GoodsList } from './components/GoodsList';
+import { goodsFromServer } from './api/goods';
+
+const App: React.FC<{}> = () => {
+  const [lengthLimit, setLengthLimit] = useState(0);
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const goods = [...goodsFromServer];
+
+  const showAll = () => {
+    setLengthLimit(-1);
+  };
+
+  const reverse = () => {
+    setIsReverse(!isReverse);
+  };
+
+  const sortByLength = () => {
+    setSortBy('length');
+  };
+
+  const sortByName = () => {
+    setSortBy('name');
+  };
+
+  const reset = () => {
+    setSortBy('');
+    setIsReverse(false);
+    setLengthLimit(-1);
+  };
+
+  const visibleGoods = goods.filter(good => {
+    if (lengthLimit === -1) {
+      return true;
+    }
+
+    return good.length <= lengthLimit;
+  });
+
+  visibleGoods.sort((f1, f2) => {
+    switch (sortBy) {
+      case 'length':
+        return f1.length - f2.length;
+      case 'name':
+        return f1.localeCompare(f2);
+      default:
+        return 0;
```

Comment:
i'd use a single function for this logic and removed usage of useState `setSortBy`